### PR TITLE
[Dev] Update LLM model wrapper mapping

### PR DESCRIPTION
**Pull request by Anna-Xiong**
## Description

- From vizro side, we added allow list of AzureOpenAI models and refactor how we can are checking available models
- However, after fixing the bug from `vizro` side, there is still issues from `langchain` side about azure open ai , the response will be 'TypeError: Completions.create() got an unexpected keyword argument 'functions'' ,
  - see issues here from langchain repo, no solution or further steps are provided as for now in those threads:
  - https://github.com/langchain-ai/langchain/issues/21881,
  - https://github.com/langchain-ai/langchain/issues/20895,
  - https://github.com/langchain-ai/langchain/issues/20013

Update:
1. We decided to merge this one first to update the model allow mechanism per our discussion @maxschulz-COL @lingyielia 
2.We  need to continue track the AzureOpenAI issues from langchain repo

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-ai/src/vizro_ai/chains/_llm_models.py [hunk 1]
```diff
@@ -1,47 +1,30 @@
...
-    },
-    "gpt-4-turbo": {
-        "max_tokens": 128000,
-        "wrapper": ChatOpenAI,
-    },
+LLM_MODELS = type(Union[ChatOpenAI, AzureOpenAI])  # Need type here to pass py3.9 unit test?
+
+SUPPORTED_MODELS = {
+    "OpenAI": [
+        "gpt-3.5-turbo-0613",
```
Comment (lingyielia) [inline vizro-ai/src/vizro_ai/chains/_llm_models.py:10]: This one is deprecated: https://platform.openai.com/docs/deprecations/2023-11-06-chat-model-updates

File: vizro-ai/src/vizro_ai/chains/_llm_models.py [hunk 2]
```diff
@@ -1,47 +1,30 @@
...
-    },
-    "gpt-3.5-turbo": {
-        "max_tokens": 16385,
-        "wrapper": ChatOpenAI,
-    },
-    "gpt-4-turbo": {
-        "max_tokens": 128000,
-        "wrapper": ChatOpenAI,
-    },
+LLM_MODELS = type(Union[ChatOpenAI, AzureOpenAI])  # Need type here to pass py3.9 unit test?
```
[the remark this answers is not in the document]
Comment (lingyielia) [reply vizro-ai/src/vizro_ai/chains/_llm_models.py:6]: With this housekeeping PR (https://github.com/mckinsey/vizro/pull/550), we could ignore my comments here.

**Comment by Anna-Xiong** [comment]
Hi @lingyielia @maxschulz-COL 
is this still relevant given the new update in main branch? https://github.com/mckinsey/vizro/pull/544

**Comment by lingyielia** [comment]
These 2 PRs are different. https://github.com/mckinsey/vizro/pull/544 is removing langchain_community.

**Comment by Anna-Xiong** [comment]
> These 2 PRs are different. #544 is removing langchain_community.

sorry the link of the pull request was copied wrong, it is https://github.com/mckinsey/vizro/pull/550

**Comment by maxschulz-COL** [comment]
> Hi @lingyielia @maxschulz-COL is this still relevant given the new update in main branch? #544

Can we discuss this in our catch-up this afternoon? Just to remind everyone what is going on?

**Review by antonymilne — APPROVED** [review]

**Pull request merged**